worlds/zillion/__init__.py

```diff
@@ -0,0 +1,374 @@
+        If you need any last-second randomization, use MultiWorld.slot_seeds[slot] instead."""
+        self.finalize_item_locations()
+
+        assert self.zz_system.patcher, "didn't get patcher from generate_early"
+        # original_rom_bytes = self.zz_patcher.rom
+        patched_rom_bytes = self.zz_system.patcher.get_patched_bytes()
+
+        out_file_base = self.world.get_out_file_name_base(self.player)
+
+        filename = os.path.join(
+            output_directory,
+            f'{out_file_base}{ZillionDeltaPatch.result_file_ending}'
+        )
+        with open(filename, "wb") as binary_file:
+            binary_file.write(patched_rom_bytes)
+        patch = ZillionDeltaPatch(
+            os.path.splitext(filename)[0] + ZillionDeltaPatch.patch_file_ending,
+            player=self.player,
+            player_name=self.world.player_name[self.player],
+            patched_path=filename
+        )
+        patch.write()
+        os.remove(filename)
+
+    def fill_slot_data(self) -> Dict[str, Any]:  # json of WebHostLib.models.Slot
+        """Fill in the `slot_data` field in the `Connected` network package.
+        This is a way the generator can give custom data to the client.
+        The client will receive this as JSON in the `Connected` response."""
+
+        # TODO: share a TypedDict data structure with client
+
+        # TODO: tell client which canisters are keywords
+        # so it can open and get those when restoring doors
+
+        zz_patcher = self.zz_system.patcher
+        assert zz_patcher, "didn't get patcher from generate_early"
+        assert self.zz_system.randomizer, "didn't get randomizer from generate_early"
+
+        rescues: Dict[str, Any] = {}
+        for i in (0, 1):
+            if i in zz_patcher.rescue_locations:
+                ri = zz_patcher.rescue_locations[i]
+                rescues[str(i)] = {
+                    "start_char": ri.start_char,
+                    "room_code": ri.room_code,
+                    "mask": ri.mask
+                }
+        zz_patcher.loc_memory_to_loc_id
+        return {
+            "start_char": self.zz_system.randomizer.options.start_char,
+            "rescues": rescues,
+            "loc_mem_to_id": zz_patcher.loc_memory_to_loc_id
+        }
+
+    def modify_multidata(self, multidata: Dict[str, Any]) -> None:
+        """For deeper modification of server multidata."""
+        # not modifying multidata, just want to call this at the end of the generation process
+        clear_cache()
+
+    # end of ordered Main.py calls
+
+    def create_item(self, name: str) -> Item:
+        """Create an item for this world type and player.
+        Warning: this may be called with self.world = None, for example by MultiServer"""
+        item_id = _item_name_to_id[name]
+
+        if not self.id_to_zz_item:
+            self._make_item_maps("JJ")
+            self.logger.warning("warning: called `create_item` without calling `generate_early` first")
```

Comment:
What's the reason to do this? I think general consent so far was that you can basically always create all items as id_to_name has to be static anyway.
(Also the server cheat console can send any item at any time. The game or client to ignore it. But that's different from create_item)

Reply:
There are 3 characters you can play with in the game.
You start the game with 1, and the other 2 are items that you can collect.
If I don't know which character I started with (information in the options during generation), then I don't know which items it's possible to collect.

The `zz_item` that is attached to the `ZillionItem` has information on collecting the item.

I will test later what happens if I `!getitem` each of the characters with each of the starting characters...

Reply:
9/9 combinations of start characters and `!getitem` characters worked as expected.